src/main/java/redis/clients/jedis/JedisSlotBasedConnectionHandler.java

```diff
@@ -44,9 +43,12 @@ public Jedis getConnection() {
         if (result.equalsIgnoreCase("pong")) return jedis;
 
         pool.returnBrokenResource(jedis);
-      } catch (JedisConnectionException ex) {
+      } catch (JedisException ex) {
         if (jedis != null) {
-          pool.returnBrokenResource(jedis);
+          try {
```

Comment:
Method returnBrokenResource() is able to throw JedisException. In bug #1168 we sometimes got exception like this:

redis.clients.jedis.exceptions.JedisException: Could not return the resource to the pool
at redis.clients.util.Pool.returnBrokenResourceObject(Pool.java:100)
at redis.clients.jedis.JedisPool.returnBrokenResource(JedisPool.java:112)
at redis.clients.jedis.JedisPool.returnBrokenResource(JedisPool.java:12)
at redis.clients.jedis.Jedis.close(Jedis.java:3355)
...
Caused by: java.lang.IllegalStateException: Invalidated object not currently part of this pool
at org.apache.commons.pool2.impl.GenericObjectPool.invalidateObject(GenericObjectPool.java:643)
at redis.clients.util.Pool.returnBrokenResourceObject(Pool.java:98)
...

I am not sure whether we can get such exception here, but throwing it further up seems inappropriate, so I added the try-catch block.